prometheus-client/CHANGELOG.md

```diff
@@ -2,6 +2,67 @@
 
 ## HEAD
 
+Version 0.4.0 is a significant rewrite of some core parts of the API, as
+`prometheus-effect` has been deprecated in favour of this library.
+
+### Breaking Changes
+
+#### New Metric API
+
+The most substatial change regards the `Metric` interface. Unregistered counters
+are now wrapped in the `Metric` type (as before), but when registered, the
+`Metric` wrapper is removed. This change means that registered metrics are more
+lightweight (they are often just a newtype wrapper), but also helps users
+distinguish between registered and unregistered metrics.
+
+To adapt to this change, users should:
+
+* Store and pass registered metrics. Rather than passing `Metric Counter`,
+  prefer `register`ing to obtain a `Counter` and pass that around.
+
+* Top level registrations will no longer return `Metric`. If you have
+
+    ```haskell
+    httpRequests :: Metric Counter
+    httpRequests = unsafeRegisterIO $ counter (Info ...)
+    ```
+
+    This should change to
+
+    ```haskell
+    httpRequests :: Metric Counter
```

Comment:
Should this be `httpRequests :: Counter` ?

Reply:
Indeed it should, thanks!